src/main/java/io/vertx/ext/jdbc/impl/actions/JDBCSQLRowStream.java

```diff
@@ -169,10 +176,9 @@ private void readRow(Future<JsonArray> fut) {
             result.addNull();
           }
         }
-        fut.complete(result);
-      } else {
-        fut.complete();
+        accumulator.add(result);
       }
+      fut.complete();
```

Comment:
add a `paused.set(false)` volatile store to create an happens before relationship with the accumulator read